Add documentation for debugging Godot Engine with CLion (#4376)
godotengine/godot-docs: development/cpp/configuring_an_ide/clion.rst

If you run into any issues, ask for help in one of
`Godot's community channels <https://godotengine.org/community>`__.

Debugging the project
---------------------

> 
> Since CLion does not support scons, you won't be able to compile, launch, and debug Godot from CLion in one step.

Reviewer: ```suggestion

Since CLion does not support SCons, you won't be able to compile, launch, and debug Godot from CLion in one step.
```

You will first need to `compile godot yourself <https://docs.godotengine.org/en/stable/development/compiling/index.html>`__ and run the binary without CLion. You will then be able to debug Godot by using the `Attach to process <https://www.jetbrains.com/help/clion/attaching-to-local-process.html>`__ feature.

> - Run the compilation with the debug flag like so : ``scons platform=x11 target=debug`` (replace x11 by your platform)

Reviewer: The platform is automatically detected now, no need to specify it. The default target is also ``target=debug``.

```suggestion
- Run the compilation in debug mode by entering ``scons``.
```

> - Run the binary you have created (in the bin directory). If you want to debug a specific project, run the binary with the following arguments : ``--editor --path path/to/your/godot/project``. To run the project instead of editing it, remove the "--editor" argument.

Reviewer: ```suggestion
- Run the binary you have created (in the bin directory). If you want to debug a specific project, run the binary with the following arguments : ``--editor --path path/to/your/godot/project``. To run the project instead of editing it, remove the ``--editor`` argument.
```

> - From CLion go to **Run > Attach to process...**

Reviewer: ```suggestion
- In CLion, go to **Run > Attach to Process...**
```

.. figure:: img/clion_4_select_attach_to_process.png
   :align: center

- Find and Select godot in the list (or type the binary name/Process ID)

.. figure:: img/clion_5_select_godot_process.png
   :align: center

You can now use the debugging tools from CLion.

> **Note** : If you run the binary without any arguments, you will only debug the project manager window. Don't forget to add the ``--path path/to/your/godot/project`` argument to debug a project.

Reviewer: ```suggestion
.. note::

    If you run the binary without any arguments, you will only debug the project manager window.
    Don't forget to add the ``--path path/to/your/godot/project`` argument to debug a project.
```
